src/devices.cpp

```diff
@@ -0,0 +1,41 @@
+#include "devices.hpp"
+
+#include <Arduino.h>
+
+extern File file;
+extern MMA8452Q accelerometer;
+extern BMP280 bmp;
+extern Adafruit_GPS gps;
+
+namespace obc {
+
+void init()
+{
+    sd_init().expect("SD init failure");
+
+    if (auto result = init(accelerometer); result.is_err()) {
+        const String msg =
+            String("Accelerometer not initialized properly, errc: ")
+            + to_underlying(result.unwrap_err());
+        file_appendln("/errors.txt", msg);
+        panic(msg.c_str());
+    }
+
+    if (auto result = init(bmp); result.is_err()) {
+        const String msg = String("Barometer not initialized properly, errc: ")
+                         + to_underlying(result.unwrap_err());
+        file_appendln("/errors.txt", msg);
+        panic(msg.c_str());
+    }
+
+    if (auto result = init(gps); result.is_err()) {
+        const String msg = String("GPS not initialized properly, errc: ")
+                         + to_underlying(result.unwrap_err());
+        file_appendln("/errors.txt", msg);
+        panic(msg.c_str());
+    }
+
+    obc::file_appendln("/boot.txt", "Devices initialized properly.");
```

Comment:
jesteśmy w namespace obc - ujednolicić zapis albo wszędzie obc::file_appendln albo bez obc::

Reply:
to nie jest do końca dobra praktyka. Rozumiem skąd się to bierze, ale podczas pisania bilbiotek, w skrajnych przypadkach, można się naciąc na bardzo ciężkie do zdebugowania problemy z ADL. Wśród społeczności deweloperów bibliotek przyjęło się, że kwalifikuje sie każde użycie funkcji namespacem.

Piszemy binarke wiec u nas nie ma to zbytnio znaczenia, ot taka ciekawa zasada

Reply:
więc jak wszędzie to wszędzie dokładać obc::